src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -20,11 +20,47 @@ public class TemplateBinder
         private readonly RouteValueDictionary _filters;
         private readonly RouteTemplate _template;
 
+        /// <summary>
+        /// <para>
+        /// This constructor is obsolete and will be removed in a future version. The recommended
+        /// alternative is the overload that does not take a UrlEncoder.
+        /// </para>
+        /// <para>Creates a new instance of <see cref="TemplateBinder"/>.</para>
+        /// </summary>
+        /// <param name="urlEncoder">The <see cref="UrlEncoder"/>.</param>
+        /// <param name="pool">The <see cref="ObjectPool{T}"/>.</param>
+        /// <param name="template">The <see cref="RouteTemplate"/> to bind values to.</param>
+        /// <param name="defaults">The default values for <paramref name="template"/>.</param>
+        [Obsolete("This constructor is obsolete and will be removed in a future version. The recommended " +
+            "alternative is the overload that does not take a UrlEncoder.")]
         public TemplateBinder(
             UrlEncoder urlEncoder,
             ObjectPool<UriBuildingContext> pool,
             RouteTemplate template,
             RouteValueDictionary defaults)
+            : this(pool, template, defaults, urlEncoder)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="TemplateBinder"/>.
+        /// </summary>
+        /// <param name="pool">The <see cref="ObjectPool{T}"/>.</param>
+        /// <param name="template">The <see cref="RouteTemplate"/> to bind values to.</param>
+        /// <param name="defaults">The default values for <paramref name="template"/>.</param>
+        public TemplateBinder(
+            ObjectPool<UriBuildingContext> pool,
+            RouteTemplate template,
+            RouteValueDictionary defaults)
+            : this(pool, template, defaults, UrlEncoder.Default)
+        {
+        }
+
+        internal TemplateBinder(
```

Comment:
Don't add this constructor, it's just the original constructor with its parameters reordered. Do main work in new constructor just above.